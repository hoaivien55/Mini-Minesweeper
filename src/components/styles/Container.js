import styled from "styled-components";

import { center } from "./sharecss";

const Container = styled.div`
	${center}
	margin: 0 auto;
	width: ${(props) => props.width || "auto"};
	min-width: ${(props) => props.minWidth || "300px"};
	min-height: ${(props) => props.minHeight || "200px"};
	padding: ${(props) => props.padding || "20px;"};
	background-color: ${(props) => props.backgroundColor || "white"};
`;

export default Container;
