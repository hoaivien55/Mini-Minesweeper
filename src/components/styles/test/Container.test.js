import renderer from "react-test-renderer";
import Container from "../Container";

it("Container: Virtual DOM snapshot", () => {
	const tree = renderer.create(<Container />).toJSON();
	expect(tree).toMatchSnapshot();
});

it("Container: change props", () => {
	const tree = renderer
		.create(
			<Container
				width="1px"
				minWidth="100px"
				minHeight="101px"
				padding="10px"
				backgroundColor="red"
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
