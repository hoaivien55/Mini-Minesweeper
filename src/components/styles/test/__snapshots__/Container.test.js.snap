// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: Virtual DOM snapshot 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  margin: 0 auto;
  width: auto;
  min-width: 300px;
  min-height: 200px;
  padding: 20px;
  background-color: white;
}

<div
  className="c0"
/>
`;

exports[`Container: change props 1`] = `
.c0 {
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  margin: 0 auto;
  width: 1px;
  min-width: 100px;
  min-height: 101px;
  padding: 10px;
  background-color: red;
}

<div
  className="c0"
  width="1px"
/>
`;
